app/task_routes.py

```diff
@@ -0,0 +1,130 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from datetime import datetime
+from app import db
+from app.models.task import Task
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_complete_request(request_body):
+    try:
+        if request_body["title"] and request_body["description"]:
+            return request_body
+
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+def validate_model_id(cls, model_id):
+    try:
+        model_id = int(model_id)    
+    except:
+        abort(make_response({"details": "Invalid data"}, 404))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details": "Invalid data"}, 404))
+    
+    return model
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    valid_data = validate_complete_request(request_body)
+    new_task = Task.from_dict(valid_data)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    task_response = {
+        "task": new_task.to_dict_post_put()
+    }
+    return make_response(jsonify(task_response), 201)
+
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks_sort_asc():
+    task_query = Task.query.all()
```

Comment:
Here, we should have `task_query = Task.query` without `all()` because we dont actually want to go query everything yet. We want to see which request params were provided to the request and then filter or order accordingly. Finally if no request params were provided then we'd want to query for all tasks near the end of this method.